memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -172,24 +179,64 @@ final case class MutableHistogram(buckets: HistogramBuckets, values: Array[Doubl
   final def copy: Histogram = MutableHistogram(buckets, values.clone)
 
   /**
-   * Adds the values from another MutableHistogram having the same bucket schema.  If it does not, then
-   * an exception is thrown -- for now.  Modifies itself.
+   * Adds the values from another Histogram.
+   * If the other histogram has the same bucket scheme, then the values are just added per bucket.
+   * If the scheme is different, then an approximation is used so that the resulting histogram has
+   * an approximate sum of the individual distributions, with the original scheme.  Modifies itself.
    */
-  final def add(other: HistogramWithBuckets): Unit =
+  final def addNoCorrection(other: HistogramWithBuckets): Unit =
     if (buckets == other.buckets) {
       // If it was NaN before, reset to 0 to sum another hist
       if (values(0).isNaN) java.util.Arrays.fill(values, 0.0)
       for { b <- 0 until numBuckets optimized } {
         values(b) += other.bucketValue(b)
       }
     } else {
-      throw new UnsupportedOperationException(s"Cannot add other with buckets ${other.buckets} to myself $buckets")
+      throw new UnsupportedOperationException(s"Cannot add histogram of scheme ${other.buckets} to $buckets")
+      // TODO: In the future, support adding buckets of different scheme.  Below is an example
+      // var ourBucketNo = 0
+      // for { b <- 0 until other.numBuckets optimized } {
+      //   // Find our first bucket greater than or equal to their bucket
+      //   while (ourBucketNo < numBuckets && bucketTop(ourBucketNo) < other.bucketTop(b)) ourBucketNo += 1
```

Comment:
If you plan to leave the code commented, please also document the problems with this approach.
IMO, it is not viable for the reasons we discussed. Primarily because it is not commutative, and it can be performed with better granularity.

Reply:
I'm adding a note.